cmd/osquery-perf/agent.go

```diff
@@ -286,14 +289,25 @@ type agent struct {
 	MDMCheckInInterval    time.Duration
 	DiskEncryptionEnabled bool
 
-	scheduledQueryData *sync.Map
+	// Note that a sync.Map is safe for concurrent use, but we still need a mutex
+	// because we read and write the field itself (not data in the map) from
+	// different goroutines (the write is in a.config).
+	scheduledQueryMapMutex sync.RWMutex
```

Comment:
This was unrelated, but I noticed the race while verifying that my changes did not introduce one.